plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1788,6 +1807,30 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                         .collect(toImmutableList())));
     }
 
+    private void removeNonCurrentQueryFiles(ConnectorSession session, Path partitionPath)
+    {
+        String queryId = session.getQueryId();
+        try {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), partitionPath);
+            RemoteIterator<LocatedFileStatus> iterator = fileSystem.listFiles(partitionPath, false);
+            while (iterator.hasNext()) {
+                Path file = iterator.next().getPath();
+                if (!isFileCreatedByQuery(file.getName(), queryId)) {
+                    fileSystem.delete(file, false);
+                }
+            }
+        }
+        catch (IOException ex) {
+            throw new TrinoException(
+                    HIVE_FILESYSTEM_ERROR,
+                    format(
+                            "Failed to delete partition %s files during overwrite in queryId: %s",
```

Comment:
nit: I think queryId is reported anyway; so no need to put that in message. But please check.

Reply:
Checked. There is error logged about Split processing error which includes queryId in its name.